cassandane/Cassandane/Test/NewTestUrl.pm

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/perl
+
+package Cassandane::Test::NewTestUrl;
+use strict;
+use warnings;
+
+use JSON;
+use LWP::UserAgent;
+
+use base qw(Cassandane::Unit::TestCase);
+
+sub new
+{
+    my $class = shift;
+    my $self = $class->SUPER::new(@_);
+    return $self;
+}
+
+sub test_basic
+{
+    my ($self) = @_;
+
+    my $string_based_url = $self->new_test_url("string based");
+    my $code_based_url = $self->new_test_url(sub {
+        return [
+            201,
+            [],
+            [ "code based" ],
+        ],
+    });
+
+    my $lwp = LWP::UserAgent->new;
+
+    {
+        my $res = $lwp->get($string_based_url->url);
+        $self->assert_str_equals('200', $res->code);
+        $self->assert_str_equals('string based', $res->decoded_content);
+    }
+
+    {
+        my $res = $lwp->get($code_based_url->url);
+        $self->assert_str_equals('201', $res->code);
+        $self->assert_str_equals('code based', $res->decoded_content);
+    }
+
+    $string_based_url->unregister;
+
+    {
+        # Unregistering one shouldn't affect the other
+        my $res = $lwp->get($code_based_url->url);
+        $self->assert_str_equals('201', $res->code);
+        $self->assert_str_equals('code based', $res->decoded_content);
+    }
+
+    $code_based_url->update("newval");
+
+    {
+        my $res = $lwp->get($code_based_url->url);
+        $self->assert_str_equals('200', $res->code);
+        $self->assert_str_equals('newval', $res->decoded_content);
+    }
+
+    {
+        eval { $string_based_url->url };
+        my $err = $@;
+        $self->assert(
+            $err =~ /\QCannot call ->url after ->unregister has been called\E/,
+        );
```

Comment:
Oh you could use `$self->assert_matches(...)` for this, which does the same thing but gets you a good error message if it fails (and the opposite, `$self->assert_does_not_match(...)`)

Reply:
Ah wasn't aware of that. I stole this pattern from another test file

Reply:
Fixed!